apps/ios/SimpleXChat/ErrorAlert.swift

```diff
@@ -11,15 +11,38 @@ import SwiftUI
 public struct ErrorAlert: Error {
     public let title: LocalizedStringKey
     public let message: LocalizedStringKey?
+    public let actionsOverride: Array<Action>?
 
-    public init(title: LocalizedStringKey, message: LocalizedStringKey?) {
+    public struct Action: Identifiable {
+        public let titleKey: LocalizedStringKey
+        public let role: ButtonRole?
+        public let action: (() -> Void)?
+
+        public var id: Int { titleKey.hashValue }
+
+        public init(
+            titleKey: LocalizedStringKey,
+            role: ButtonRole? = nil,
+            action: (() -> Void)? = { }
+        ) {
+            self.titleKey = titleKey
+            self.role = role
+            self.action = action
+        }
+    }
+
+    public init(
+        title: LocalizedStringKey,
+        message: LocalizedStringKey? = nil,
+        actionsOverride: Array<Action>? = nil
+    ) {
```

Comment:
So this adds support for adding custom actions to `ErrorAlert`.
Required, since network timeout error needs to be cancellable.

Reply:
and we are one step closer to just sending Alert around...